icann-rdap-client/src/md/redacted.rs

```diff
@@ -61,3 +68,501 @@ impl ToMd for &[Redacted] {
+                                    }
+                                    _ => {
+                                        // Do nothing but we need to investigate why this is happening
+                                    }
+                                } // end match replace_with
+                            } // end if doing partialValue or emptyValue
+                        } // end if redaction_type
+                    } // end if final_path_option
+                } // end for each path_index_count
+            } // end if final_path
+        } // end loop over redactions
+    } // end if there is a redacted array
+}
+
+// This cleans it up into a json pointer which is what we need to use to get the value
+fn convert_to_json_pointer_path(path: &str) -> String {
+    let pointer_path = path
+        .trim_start_matches('$')
+        .replace('.', "/")
+        .replace("['", "/")
+        .replace("']", "")
+        .replace('[', "/")
+        .replace(']', "")
+        .replace("//", "/");
+    pointer_path
+}
+
+fn parse_redacted_array(
+    rdap_json_response: &Value,
+    redacted_array: &Vec<Value>,
+) -> Vec<RedactedObject> {
+    let mut list_of_redactions: Vec<RedactedObject> = Vec::new();
+
+    for item in redacted_array {
+        let item_map = item.as_object().unwrap();
+        let pre_path = item_map
+            .get("prePath")
+            .and_then(|v| v.as_str())
+            .map(|s| s.to_string());
+        let post_path = item_map
+            .get("postPath")
+            .and_then(|v| v.as_str())
+            .map(|s| s.to_string());
+        // this is the original_path given to us
+        let original_path = pre_path.clone().or(post_path.clone());
+        let mut redacted_object = RedactedObject {
+            name: Value::String(String::from("")), // Set to empty string initially
+            path_index_count: 0,                   // Set to 0 initially
+            pre_path,
+            post_path,
+            original_path,
+            final_path: Vec::new(), // final path we are doing something with
+            do_final_path_subsitution: false, // flag whether we ACTUALLY doing something or not
+            path_lang: item_map
+                .get("pathLang")
+                .unwrap_or(&Value::String(String::from("")))
+                .clone(),
+            replacement_path: item_map
+                .get("replacementPath")
+                .and_then(|v| v.as_str())
+                .map(|s| s.to_string()),
+            method: item_map
+                .get("method")
+                .unwrap_or(&Value::String(String::from("")))
+                .clone(),
+            reason: Value::String(String::from("")), // Set to empty string initially
+            result_type: Vec::new(), // Set to an empty Vec<Option<ResultType>> initially
+            redaction_type: None,    // Set to None initially
+        };
+
+        // Check if the "name" field is an object
```

Comment:
Wouldn't it be cleaner to create a struct for "name" that has an enum of "description" and "type"?

Reply:
All I care about for the RedactedObject name field is that it is useful for debugging purposes. I can't see any other use for the type/description so I "airlifted" it up one level.